### DISCUSSION TITLE
Naming of liters: liter/L or litre/l

### DISCUSSION THREAD

**Discussion by angularsen** [angularsen/UnitsNet#1201] [Polls]
See #1115 for some more background.

- Option 1 - liter/L - No change. Consistent with American standards and language.
- Option 2 - liter/l - Change "L" to "l". No breaking change (1), but also not consistent with any industry standard (2).
- Option 3 - litre/l - Change both. Consistent with international industry standards, breaking change for many (3).

1) Changing case of unit abbreviations is generally not a breaking change, parsing is case-insensitive if not ambiguous.

2) American standards use liter/L while international and European standards use litre/l: https://github.com/angularsen/UnitsNet/issues/1115#issuecomment-1374479034 

3) Breaking change due to unit enum names like `VolumeUnit.Liter` being changed to `VolumeUnit.Litre`. We can soften the transition by having both for a period of time, marking the old one as obsolete so compile warnings give instructions to change it, before removing it in the next major version.

This discussion can be extended to, should we generally name units according to international or American industry standards?

**Comment by jblam** [comment]
> should we generally name units according to international or American industry standards

IMO take the breaking change and switch everything to the correct, standardised names e.g. "metre".

**Reply by angularsen** [reply]
I understand the point of view, but I don't want to incur breaking changes "just because".

In my opinion, the greater majority would want this change or at least care about it for it to be worth requiring everyone to make changes to their code when they upgrade.

Consensus from the few opinions gathered so far seems to be that "litre" and "metre" does not carry much weight.
I guess most are used to reading both variants and for better or worse it seems to me American English is maybe more widely used in the programming domain.

If there was a strong signal that most wanted this change, I would happily make it.
But currently it sits around 50/50 and I am not convinced it is worth it.

I'll leave this poll open for future reference to potentially gather more votes, but my current position is to make no change to litre/liter.

As for "L" vs "l", there is some work in progress to test out making it either configurable or based on culture. I think this could be a good compromise.
https://github.com/angularsen/UnitsNet/pull/1209
